tests/llm/test_tracking.py

```diff
@@ -0,0 +1,180 @@
+import os
+import time
+from functools import partial
+
+import pytest
+
+import agentlab.llm.tracking as tracking
+
+
+def test_get_action_decorator():
+    action, agent_info = tracking.get_action_decorator(lambda x, y: call_llm())(None, None)
+    assert action == "action"
+    assert agent_info["stats"] == {
+        "input_tokens": 1,
+        "output_tokens": 1,
+        "cost": 1.0,
+    }
+
+
+OPENROUTER_API_KEY_AVAILABLE = os.environ.get("OPENROUTER_API_KEY") is not None
+
+OPENROUTER_MODELS = (
+    "openai/o1-mini-2024-09-12",
+    "openai/o1-preview-2024-09-12",
+    "openai/gpt-4o-2024-08-06",
+    "openai/gpt-4o-2024-05-13",
+    "anthropic/claude-3.5-sonnet:beta",
+    "anthropic/claude-3.5-sonnet",
+    "meta-llama/llama-3.1-405b-instruct",
+    "meta-llama/llama-3.1-70b-instruct",
+    "meta-llama/llama-3.1-8b-instruct",
+    "google/gemini-pro-1.5",
+    "qwen/qwen-2-vl-72b-instruct",
+)
+
+
+@pytest.mark.skipif(not OPENROUTER_API_KEY_AVAILABLE, reason="OpenRouter API key is not available")
+def test_get_pricing_openrouter():
+    pricing = tracking.get_pricing(api="openrouter", api_key=os.environ["OPENROUTER_API_KEY"])
+    assert isinstance(pricing, dict)
+    assert all(isinstance(v, dict) for v in pricing.values())
+    for model in OPENROUTER_MODELS:
+        assert model in pricing
+        assert isinstance(pricing[model], dict)
+        assert all(isinstance(v, float) for v in pricing[model].values())
+
+
+def test_get_pricing_openai():
+    pricing = tracking.get_pricing(api="openai")
+    assert isinstance(pricing, dict)
+    assert all("prompt" in pricing[model] and "completion" in pricing[model] for model in pricing)
+    assert all(isinstance(pricing[model]["prompt"], float) for model in pricing)
+    assert all(isinstance(pricing[model]["completion"], float) for model in pricing)
```

Comment:
@recursix I was thinking of testing for specific values but i feel like it might be very inconsistent, do you have an opinion on that matter ?